packages/node-http-handler/src/node-http2-handler.ts

```diff
@@ -0,0 +1,111 @@
+import { connect, constants, ClientHttp2Session } from "http2";
+
+import { Readable } from "stream";
+import { buildQueryString } from "@aws-sdk/querystring-builder";
+import {
+  HttpHandler,
+  HttpHandlerOptions,
+  HttpRequest,
+  HttpResponse,
+  NodeHttpOptions
+} from "@aws-sdk/types";
+
+import { writeRequestBody } from "./write-request-body";
+import { getTransformedHeaders } from "./get-transformed-headers";
+
+export class NodeHttp2Handler
+  implements HttpHandler<Readable, NodeHttpOptions> {
+  private readonly connectionPool: Map<string, ClientHttp2Session>;
+
+  constructor(private readonly httpOptions: NodeHttpOptions = {}) {
```

Comment:
Do we need a `NodeHttp2Options`? Is the Http1 and Http2 client option interfaces always the same?

Reply:
I'd created NodeHttp2Options to create sessionTimeout variable.
But reused socketTimeout instead, as both of them mean the same ([documentation](https://nodejs.org/api/http2.html#http2_http2session_and_sockets))

http2 doesn't have keepAlive though

Reply:
Resolved in 480dd1bee93c725061661faf7bc6af2c42ed099c